src/github.com/matrix-org/dendrite/clientapi/auth/auth.go

```diff
@@ -27,6 +29,13 @@ import (
 	"github.com/matrix-org/util"
 )
 
+// UnknownDeviceID is the default device id if one is not specified.
+var UnknownDeviceID = "unknown-device"
```

Comment:
It looks like synapse generates a random device ID if one isn't specified?
https://github.com/matrix-org/synapse/blob/master/synapse/handlers/device.py#L82